x-pack/plugins/cases/public/components/configure_cases/index.tsx

```diff
@@ -334,10 +336,42 @@ export const ConfigureCases: React.FC = React.memo(() => {
     (data: CustomFieldConfiguration) => {
       const updatedCustomFields = addOrReplaceField(customFields, data);
 
+      // add the new custom field to each template as well
+      const updatedTemplates = updatedCustomFields.length
```

Comment:
If `data` cannot be undefined isn't this check `updatedCustomFields.length ? ... : templates;` redundant? there is always going to be `updatedCustomFields`.